src/map/script.cpp

```diff
@@ -24770,24 +24770,31 @@ BUILDIN_FUNC(hateffect){
 		return SCRIPT_CMD_FAILURE;
 	}
 
-	auto it = util::vector_get( sd->hatEffects, effectID );
+	unit_data* ud = unit_bl2ud( bl );
+
+	if( ud == nullptr ){
+		ShowError( "buildin_hateffect: unsupported unit type %d\n", bl->type );
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	auto it = util::vector_get( ud->hatEffects, effectID );
 
 	if( enable ){
-		if( it != sd->hatEffects.end() ){
+		if( it != ud->hatEffects.end() ){
```

Comment:
Unrelated to the PR but `vector_exists` could be used instead of `vector_get`

Reply:
Fixed